railgun/keps/0006-officially-supported-platforms.md

```diff
@@ -0,0 +1,54 @@
+---
+status: provisional
+---
+
+# Official Support for new Platforms and Integrations with KIC
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+<!-- /toc -->
+
+## Summary
+
+Provide the criteria for contributors to request and add new "official supported" platforms and integrations for the [Kong Kubernetes Ingress Controller (KIC)][kic].
+
+[kic]:https://github.com/kong/kubernetes-ingress-controller
+
+## Motivation
+
+- we want to document for contributors and end-users the criteria for any platform or integration to be "officially supported"
+- we want to demonstrate our documentation by adding new support for an existing feature request using the newly established criteria
+
+### Goals
+
+- establish preliminary acceptance criteria, including considerations about impact and maintenance burden
+- establish testing and infrastructure criteria
+- document the timeframes relevant to any official support, with clear end of life (EOL) criteria
+- provide walkthrough documentation that covers beginning to end
+- provide a repository issue template for this purpose which links to the relevant documentation
+- exercise and demonstrate all of the above by implementing the [previously requested ARM64 support][issues-451]
+
+[arm64]:https://en.wikipedia.org/wiki/ARM64
+[upstream]:https://github.com/kong/kong
+[issues-451]:https://github.com/Kong/kubernetes-ingress-controller/issues/451
+
```

Comment:
> In a zoom call with @shaneutt we've identified smaller problems that are potentially solvable with this KEP:
> 
>     1. Make KIC support arm64
>     2. Use the process (being built here) to inform the decision whether we want arm64 support or not
>     3. Define criteria based on which KIC maintainers will be able to call a platform X or an integration with Y "supported"
>     4. Improve the experience for (communicate properly with) community users asking for support of platform X or an integration with Y
>     5. Define criteria by which KIC maintainers will decide whether to commit to supporting platform X / integration with Y
> 
> TBD: which of these motivations we want to take care of

In our zoom earlier we discussed how number 5 here is ultimately the contentious one and we had alignment on the remainder. To bridge the gap I suggest this change in language: 

```suggestion

# Non Goals

As a result of this proposal we want to explicitly stay away from having processes that are developed out of this become authoritative "accepted"/"declined" processes in favor of them being prescriptive to help contributors, but if we're going to come up with a process for hard declines or hard committing to features that should be a later iteration after we've had some time to see the effects of these changes.
```

Reply:
lmkwyt @mflendrich ?